### PR TITLE
executor: add spawner

### LINKED ISSUE #1: Allow spawning task from outside an Executor
With smol 0.2 and below, we could spawn tasks from anywhere, starting with smol 0.3 which uses async-executor, it's no longer possible. It makes it impossible to spawn tasks from e.g. a custom reactor thread which doesn't run on the Executor.
This seem to be a pretty sad regression.

It would be great to at least have some king of handle to spawn tasks on the executor from the outside?

### PR DESCRIPTION
**Pull request by Keruspe**
This allows spawning a task on an executor from outside of it

Fixes #1

### CODE REVIEW

**Comment by stjepang** [comment]
Would this also need the `LocalSpawner` counterpart?

Perhaps instead of `Task::<()>::spawner()` it would make more sense to have `Spawner::get()` or `Spawner::current()` :)

Does this really require changes to `multitask`? We could have:

```rust
pub struct Executor {
    ex: Arc<multitask::Executor>,
}

pub struct Spawner {
    ex: Arc<multitask::Executor>,
}
```

**Comment by Keruspe** [comment]
Sure, don't know why I didn't sync of the Arc<Executor> version :)

I did the Task::spawner to be symmetrical with Task::spawn but `Spawner::current()` looks a lot like the `Handle::current()` from tokio, fine by me

**Comment by stjepang** [comment]
Another argument for `current()` is that it matches https://doc.rust-lang.org/nightly/std/thread/fn.current.html

**Comment by Keruspe** [comment]
Wrt the LocalSpawner counterpart, I'm not sure if that's necessary. We can add it for symmetry, but if you want to spawn something on the local executor, thus on the current thread, there's a huge change you're already in it right?

File: src/lib.rs [hunk 1]
```diff
@@ -184,6 +201,62 @@ impl Default for Executor {
...
 }
 
+/// A spawner for a multi-threaded executor.
+#[derive(Debug)]
+pub struct Spawner {
+    ex: Arc<multitask::Executor>,
+}
+
+impl Spawner {
+    /// Get a Spawner for the current multi-threaded executor.
```
Comment (stjepang) [inline src/lib.rs:211]:
```suggestion
    /// Gets a spawner for the current multi-threaded executor.
```

File: src/lib.rs [hunk 2]
```diff
@@ -184,6 +201,62 @@ impl Default for Executor {
...
+    ///
+    /// If called from an [`Executor`], get its Spawner.
```
Comment (stjepang) [inline src/lib.rs:213]:
```suggestion
    /// If called from an [`Executor`], retruns its [`Spawner`].
```

**Comment by stjepang** [comment]
I guess you're right, `LocalSpawner` would be kind of useless :)

**Comment by Keruspe** [comment]
Fixed the comments

**Comment by stjepang** [comment]
Published in v0.1.2

**Comment by Keruspe** [comment]
Thanks!

Do you think smol will reexport this or will this stay in the async_executor namespace?

**Comment by stjepang** [comment]
I think it will stay.

**Pull request merged**